hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/HazelcastClientCachingProvider.java

```diff
@@ -123,6 +127,27 @@ private HazelcastInstance getDefaultInstance() {
         return hazelcastInstance;
     }
 
+    /**
+     * Get an existing {@link HazelcastInstance} by {@code instanceName} or, if not found, create a new {@link HazelcastInstance}
+     * with default configuration and given {@code instanceName}.
+     *
+     * @param instanceName name by which to lookup existing {@link HazelcastInstance} or create new one.
+     * @return             a {@link HazelcastInstance} with the given {@code instanceName}
+     */
+    private HazelcastInstance getOrCreateByInstanceName(String instanceName) {
+        HazelcastInstance instance = HazelcastClient.getHazelcastClientByName(instanceName);
```

Comment:
In case of a race, one operation will succeed while the other one will fail with a `DuplicateInstanceNameException` [here](https://github.com/vbekiaris/hazelcast/blob/77ee1969666d385b1bcbc6a6294911f398b7e9fd/hazelcast-client/src/main/java/com/hazelcast/client/HazelcastClientManager.java#L83).